lua/rustaceanvim/config/server.lua

```diff
@@ -22,13 +22,15 @@ function server.load_rust_analyzer_settings(project_root, opts)
   local default_settings = opts.default_settings or config.server.default_settings
   local use_clippy = config.tools.enable_clippy and vim.fn.executable('cargo-clippy') == 1
   ---@diagnostic disable-next-line: undefined-field
-  if default_settings['rust-analyzer'].checkOnSave == nil and use_clippy then
+  if default_settings['rust-analyzer'].check == nil and use_clippy then
     ---@diagnostic disable-next-line: inject-field
-    default_settings['rust-analyzer'].checkOnSave = {
+    default_settings['rust-analyzer'].check = {
       allFeatures = true,
       command = 'clippy',
       extraArgs = { '--no-deps' },
     }
+    ---@diagnostic disable-next-line: inject-field
+    default_settings['rust-analyzer'].checkOnSave = true
```

Comment:
issue: we should add a check for `type(default_settings['rust-analyzer'].checkOnSave) ~= 'boolean'`, so that we don't override it if it's been set in the user config.

Reply:
👍  This look better?